### PR TITLE
[TR] PIM-4536: Forbid to add values for an in-existing couple Channel / Locale

### PR DESCRIPTION
**Pull request by grena**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y
| Specs & Static     | Y
| Changelog updated  | N

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy-field.js [hunk 1]
```diff
@@ -10,15 +10,15 @@
...
     ],
-    function (Backbone, _, template, i18n) {
-        return Backbone.View.extend({
+    function (Backbone, _, Field, template, i18n, mediator) {
+        return Field.extend({
             tagName: 'div',
             field: null,
-            locale: null,
-            scope: null,
             value: {},
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy-field.js]: not used anymore ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy-field.js [hunk 2]
```diff
@@ -66,30 +70,33 @@ define([
...
-             * Set the scope
+             * Set the value to be displayed in the copy field
              *
-             * @param {string} scope
+             * @param {Object} value
              */
-            setScope: function (scope) {
-                this.scope = scope;
+            setValue: function (value) {
+                this.value = value;
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy-field.js]: not used anymore ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/locale-specific.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+                return this;
+            },
+
+            /**
+             * Update the given field by adding element to it
+             *
+             * @param {Object} field
+             */
+            updateInput: function (field) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/locale-specific.js]: The naming is bad imho, what about `updateFieldElements` ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/localizable.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            channelHasLocale: function (channel, locale) {
+                return _.contains(channel.locales, locale);
+            },
+
+            /**
+             * Update the given field by adding element to it
+             *
+             * @param {Object} field
+             */
+            updateInput: function (context, field) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/localizable.js]: same

**Comment by phaseinducer** [comment]
No new behats ?

**Comment by sumbobyboys** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM
nice refactoring of copy field btw : )

**Pull request merged**